src/components/ClusterNodes/NodeDetails.tsx

```diff
@@ -0,0 +1,765 @@
+        if (!nodeDetail) return null
+        return (
+            <div className="node-details-container">
+                <div className="ml-20 mr-20 mb-12 mt-16 pl-20 pr-20 pt-16 pb-16 bcn-0 br-4 en-2 bw-1">
+                    <div className="fw-6 fs-16 cn-9">{nodeDetail.name}</div>
+                    <div className={`fw-6 fs-13 ${TEXT_COLOR_CLASS[nodeDetail.status] || 'cn-7'}`}>
+                        {nodeDetail.status}
+                    </div>
+                </div>
+                <div className="ml-20 mr-20 mt-12 node-details-grid">
+                    <div className="fw-6 fs-16 cn-9">
+                        {renderErrorOverviewCard()}
+                        {renderProbableIssuesOverviewCard()}
+                        {renderNodeOverviewCard()}
+                    </div>
+                    <div>
+                        {renderResourceList()}
+                        {renderLabelAnnotationTaint()}
+                        {renderPodList()}
+                    </div>
+                </div>
+            </div>
+        )
+    }
+
+    const cancelYAMLEdit = (): void => {
+        setIsReviewStates(false)
+        setModifiedManifest(YAML.stringify(nodeDetail.manifest))
+    }
+
+    const handleEditorValueChange = (codeEditorData: string): void => {
+        setModifiedManifest(codeEditorData)
+    }
+
+    const saveYAML = (): void => {
+        if (isReviewState) {
+            const requestData: UpdateNodeRequestBody = {
+                clusterId: +clusterId,
+                name: nodeName,
+                manifestPatch: JSON.stringify(YAML.parse(modifiedManifest)),
+                version: nodeDetail.version,
+                kind: nodeDetail.kind,
+            }
+            setApiInProgress(true)
+            updateNodeManifest(clusterId, nodeName, requestData)
+                .then((response: NodeDetailResponse) => {
+                    setApiInProgress(false)
+                    if (response.result) {
+                        toast.success('Node updated')
+                        nodeDetail.manifest = response.result
+                        setIsReviewStates(false)
+                    }
+                })
+                .catch((error) => {
+                    showError(error)
+                    setApiInProgress(false)
+                })
+        } else {
+            setApiInProgress(true)
+            //setIsReviewStates(true)
+        }
+    }
+
+    const renderYAMLEditor = (): JSX.Element => {
+        return (
+            <div className="node-details-container">
+                <CodeEditor
+                    value={modifiedManifest}
+                    defaultValue={nodeDetail?.manifest && YAML.stringify(nodeDetail?.manifest)}
+                    height={isReviewState ? 'calc( 100vh - 170px)' : 'calc( 100vh - 137px)'}
+                    diffView={isReviewState}
+                    onChange={handleEditorValueChange}
+                    noParsing
```

Comment:
add `mode={Modes.YAML}` or it'll give some errors & will not render the colors properly.

Reply:
Done